nautobot/core/templates/inc/nav_menu.html

```diff
@@ -178,7 +205,18 @@
 
         // Save the scroll position when the navbar is scrolled
         navbar.addEventListener('scroll', debounce(function() {
-            localStorage.setItem('navbarScrollPosition', navbar.scrollTop);
+            sessionStorage.setItem('navbarScrollPosition', navbar.scrollTop);
         }, 250));
+
+         // Add click event listeners to dropdown links
+        addLinkClickListeners();
+
+        // Apply the 'active' class to the previously clicked link
+        if (activeLink) {
+            var previouslyClickedLink = document.querySelector('.navbar-fixed-left .navbar-nav > .dropdown > .nav-dropdown-menu > li > a[href="' + activeLink + '"]');
+            if (previouslyClickedLink) {
+                previouslyClickedLink.parentElement.classList.add('active');
+            }
+        }
```

Comment:
Does this risk causing confusion if the user navigates to other pages through inter- or intra-page links rather than the nav menu? To me it feels like it'd be weird for "Circuit Types" to stay active in the side bar after I click to the detail view of a specific CircuitType, and from there to the detail view of a Tenant that owns a specific circuit of this type.

Reply:
Yep - good catch! Modified it to also check that the link's href matches the current page URL:

```
// Apply the 'active' class to the previously clicked link
if (activeLink) {
    var previouslyClickedLink = document.querySelector('.navbar-fixed-left .navbar-nav > .dropdown > .nav-dropdown-menu > li > a[href="' + activeLink + '"]');
    var currentLocation = window.location.pathname + window.location.search;
    
    if (previouslyClickedLink && currentLocation.includes(previouslyClickedLink.getAttribute('href'))) {
        previouslyClickedLink.parentElement.classList.add('active');
    }
}
```